pkg/segment/query/querystatus.go

```diff
@@ -66,17 +70,40 @@ type QueryStateChanData struct {
 	Error           error // Only used when the state is ERROR
 }
 
+type WaitStateData struct {
+	qid    uint64
+	rQuery *RunningQueryState
+}
+
 const (
-	RUNNING      QueryState = iota + 1
-	QUERY_UPDATE            // flush segment counts & aggs & records (if matched)
+	READY QueryState = iota + 1
+	RUNNING
+	QUERY_UPDATE // flush segment counts & aggs & records (if matched)
 	COMPLETE
 	CANCELLED
 	TIMEOUT
 	ERROR
 )
 
+func InitMaxRunningQueries() {
+	memConfig := config.GetMemoryConfig()
+	totalMemoryInBytes := config.GetTotalMemoryAvailable()
+	searchMemoryInBytes := (totalMemoryInBytes * memConfig.SearchPercent) / 100
+	maxConcurrentQueries := searchMemoryInBytes / memConfig.MemoryPerQueryInBytes
+	if maxConcurrentQueries < 1 {
+		log.Warnf("System does not have sufficient memory for query. Setting maxConcurrentQueries to 1, searchMemoryInBytes: %v, MemoryPerQueryInBytes: %v",
+			searchMemoryInBytes, memConfig.MemoryPerQueryInBytes)
+		maxConcurrentQueries = 1
+	}
```

Comment:
we should remove this check/error and if the maxConcurreQuery num comes out to be less than 2, then set it 2

Reply:
addressed in the latest commit.